packages/terra-folder-tree/src/subcomponents/FolderTreeItem.jsx

```diff
@@ -112,6 +123,48 @@ const FolderTreeItem = ({
     }
   };
 
+  const handleKeyDown = event => {
+    switch (event.nativeEvent.keyCode) {
+      case KeyCode.KEY_RETURN:
```

Comment:
I don't think the Return key is supposed to select an input field.

Reply:
No that's a requirement of the story. It needs to perform the default action on the current item, which should be to select it